src/updateIds/updateIds.js

```diff
@@ -13,82 +13,99 @@ const { parseTest, parseSuite, replaceSuiteTitle } = require('./helpers');
+
+        const currentSuiteId = parseSuite(suiteIndex);
+        if (
+          currentSuiteId &&
+          (testomatioMap.suites[suiteIndex] !== `@S${currentSuiteId}` ||
+            testomatioMap.suites[suiteWithoutTags] !== `@S${currentSuiteId}`)
+        ) {
+          debug(`   Previous ID detected in suite '${suiteIndex}'`);
+          duplicateSuites++;
+          continue;
+        }
+
+        if (!processedSuites.has(suiteIndex)) {
+          if (testomatioMap.suites[suiteIndex] && !suite.includes(testomatioMap.suites[suiteIndex])) {
+            fileContent = replaceSuiteTitle(suite, `${suite} ${testomatioMap.suites[suiteIndex]}`, fileContent);
+            fs.writeFileSync(file, fileContent);
+            processedSuites.add(suiteIndex);
+          } else if (
+            testomatioMap.suites[suiteWithoutTags] &&
+            !suite.includes(testomatioMap.suites[suiteWithoutTags])
+          ) {
+            fileContent = replaceSuiteTitle(suite, `${suite} ${testomatioMap.suites[suiteWithoutTags]}`, fileContent);
+            fs.writeFileSync(file, fileContent);
+            processedSuites.add(suiteIndex);
+          }
+        }
       }
     }
 
     for (const test of testArr) {
-      let testIndex = `${test.suites[0] || ''}#${test.name}`;
-      debug('testIndex', testIndex);
-
-      // this is not test; its test.skip() annotation inside a test
-      if (opts.framework === 'playwright' && test.name === true) continue;
-
-      let testWithoutTags = `${(test.suites[0] || '').replace(TAG_REGEX, '').trim()}#${test.name.replace(
-        TAG_REGEX,
-        '',
-      )}`.trim();
-
-      if (!testomatioMap.tests[testIndex] && !testomatioMap.tests[testWithoutTags]) {
-        testIndex = test.name; // if no suite title provided
-        testWithoutTags = test.name.replace(TAG_REGEX, '').trim();
-      }
-
-      const currentTestId = parseTest(testIndex);
-      if (
-        currentTestId &&
-        testomatioMap.tests[testIndex] !== `@T${currentTestId}` &&
-        testomatioMap.tests[testWithoutTags] !== `@T${currentTestId}`
-      ) {
-        debug(`Previous ID detected in test '${testIndex}'`);
-        duplicateTests++;
+      if (!test) {
+        debug('Invalid test:', test);
         continue;
       }
-
-      if (testomatioMap.tests[testIndex] && !test.name.includes(testomatioMap.tests[testIndex])) {
-        fileContent = replaceAtPoint(fileContent, test.updatePoint, ` ${testomatioMap.tests[testIndex]}`);
-        fs.writeFileSync(file, fileContent);
-        delete testomatioMap.tests[testIndex];
-      } else if (testomatioMap.tests[testWithoutTags] && !test.name.includes(testomatioMap.tests[testWithoutTags])) {
-        fileContent = replaceAtPoint(fileContent, test.updatePoint, ` ${testomatioMap.tests[testWithoutTags]}`);
-        fs.writeFileSync(file, fileContent);
-        delete testomatioMap.tests[testWithoutTags];
+      const testName = test.name || '';
+
+      if (testName) {
```

Comment:
replace `if (test.name)` with `if (!test.name) continue`

and you will get rid of 1 level of nesting